interaction_resume/models/abstract_interaction_source.py

```diff
@@ -0,0 +1,95 @@
+from odoo import api, fields, models
+
+
+class InteractionSource(models.AbstractModel):
+    _name = "interaction.source"
+
+    # Just making sure the fields exist in the model
+    partner_id = fields.Many2one("res.partner")
+    date = fields.Datetime()
+
+    @api.model
+    def search_interaction(
+        self,
+        partner,
+        direction=None,
+        medium=None,
+        search_input=None,
+        since=None,
+        until=None,
+    ):
+        """
+        Search interactions for a given partner
+        @param partner:
+        @param direction:
+        @param medium:
+        @param search_input:
+        @param since:
+        @param until:
+        @return: total count when no filtering applied, interaction.resume records
+        """
+        search_domain = self._get_interaction_partner_domain(partner)
+        if direction:
+            search_domain.extend(
+                self._get_interaction_direction_domain(partner, direction)
+            )
+        if medium:
+            search_domain.extend(self._get_interaction_medium_domain(medium))
+        if search_input:
+            search_domain.extend(self._get_interaction_body_domain(search_input))
+        total = self._get_interaction_total(search_domain)
+        records = self.search(
+            [
+                *search_domain,
+                ("date", ">=", since),
+                ("date", "<=", until),
+            ]
+        )
+        interaction = self.env["interaction.resume"].create(
+            records._get_interaction_data()
+        )
+        return total, interaction
+
+    def _get_interaction_partner_domain(self, partner):
+        return [
+            "|",
+            "|",
+            ("partner_id", "=", partner.id),
+            ("partner_id.email", "=", partner.email),
+            ("partner_id", "in", partner.other_contact_ids.ids),
+        ]
+
+    def _get_interaction_total(self, search_domain):
+        return self.search_count(search_domain)
+
+    def _get_interaction_direction_domain(self, partner, direction):
+        # Should be implemented by the concrete model
```

Comment:
don't know if it's a good idea, but if we really want the users to implement this method (so that they don't have weird behaviors) should we not raise a `NotImplementedException` ?